cloud/endagaweb/views/dashboard.py

```diff
@@ -885,28 +908,404 @@ def _search_events(self, profile, query_string, orig_events):
             Returns:
                 a QuerySet that matches the query string
             """
-            network = profile.network
-            queries = query_string.split()
-
-            res_events = UsageEvent.objects.none()
-            for query in queries:
-                events = orig_events
-                events = (events.filter(kind__icontains=query)
-                          | events.filter(reason__icontains=query)
-                          | events.filter(subscriber__name__icontains=query)
-                          | events.filter(subscriber__imsi__icontains=query)
-                          | events.filter(subscriber_imsi__icontains=query))
-
-                # Get any numbers that match, and add their associated
-                # subscribers' events to the results
-                potential_subs = (
-                    Number.objects.filter(number__icontains=query)
-                                  .values('subscriber')
-                                  .filter(subscriber__network=network)
-                                  .distinct())
-                if potential_subs:
-                    events |= (UsageEvent.objects
-                                .filter(subscriber__in=potential_subs))
-
-                res_events |= events
-            return res_events
+        network = profile.network
+        queries = query_string.split()
+
+        res_events = UsageEvent.objects.none()
+        for query in queries:
+            events = orig_events
+            events = (events.filter(kind__icontains=query)
+                      | events.filter(reason__icontains=query)
+                      | events.filter(subscriber__name__icontains=query)
+                      | events.filter(subscriber__imsi__icontains=query)
+                      | events.filter(subscriber_imsi__icontains=query))
+
+            # Get any numbers that match, and add their associated
+            # subscribers' events to the results
+            potential_subs = (
+                Number.objects.filter(number__icontains=query)
+                              .values('subscriber')
+                              .filter(subscriber__network=network)
+                              .distinct())
+            if potential_subs:
+                events |= (UsageEvent.objects
+                           .filter(subscriber__in=potential_subs))
+
+            res_events |= events
+        return res_events
+
+
+class UserManagement(ProtectedView):
```

Comment:
Own file means user.py?. If yes, we can but it requires a little time for refactoring of the code.